warehouse/manage/views.py

```diff
@@ -958,7 +958,19 @@ def create_macaroon(self):
 
         response = {**self.default_response}
         if form.validate():
-            macaroon_caveats = [{"permissions": form.validated_scope, "version": 1}]
+            if form.validated_scope == "user":
+                macaroon_caveats = [{"permissions": form.validated_scope, "version": 1}]
+            else:
+                project_ids = [
+                    str(project.id)
+                    for project in self.request.user.projects
+                    if project.normalized_name in form.validated_scope["projects"]
+                ]
```

Comment:
Flagging for review: this is slightly duplicated with a check against `validated_scope` below, for project security events. I could try to deduplicate the two, but this results in the smallest/most readable diff IMO.